src/package.ts

```diff
@@ -274,6 +276,32 @@ class ManifestProcessor extends BaseProcessor {
 		}
 	}
 
+	async onFile(file: IFile): Promise<IFile> {
+		const path = util.normalize(file.path);
+
+		if (!/^extension\/package.json$/i.test(path)) {
+			return Promise.resolve(file);
+		}
+
+		// Ensure that package.json is writable as VS Code needs to
+		// store metadata in the extracted file.
+		const mode = 0o100644;
+
+		if (isInMemoryFile(file)) {
+			return {
+				path: file.path,
+				mode,
+				contents: file.contents,
+			};
+		} else {
+			return {
+				path: file.path,
+				mode,
+				localPath: file.localPath,
+			};
+		}
```

Comment:
Would this work too?

```suggestion
		return { ...file, mode };
```

Reply:
It does, thanks!